cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -156,35 +166,44 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                                            endPoint, userParams.toString());
           setErrorResponse(response, "", errorResp, SC_BAD_REQUEST, wantJSON(request));
         } else {
+          if (EndPoint.getEndpoint().contains(endPoint)) {
+            _requestMeter.get(endPoint).mark();
+          } else {
+            throw new UserRequestException("Invalid URL for GET");
```

Comment:
Why do we need this check here? Can't we move `_requestMeter.get(endPoint).mark();` within `switch-case` to avoid duplicate code here?

Reply:
Actually that is my original way of implementing and I feel it is more verbose, since we need to have the same `_requestMeter.get(endPoint).mark()` in each switch branch. I did think about removing else here since it looks redundant with the default branch of switch statement. but it actually helps use to jump out of execution and return a little bit earlier.

Reply:
Thanks for the clarification. How about (1) keeping the `if` part, and (2) removing just the `else` here? Wouldn't switch case already throws an exception if the endpoint is invalid?